### PR TITLE
file `exists` intrinsic in c backend

### PR DESCRIPTION
**Pull request by WaelYoussfi**

### CODE REVIEW

**Comment by fridis** [comment]
can you resolve the conflicts?

File: src/dev/flang/be/c/Intrinsics.java [hunk 1]
```diff
@@ -143,7 +143,20 @@ public class Intrinsics extends ANY
             );
         }
         );
-    put("fuzion.std.fileio.exists"       , noFileIo); // NYI: #158
+    put("fuzion.std.fileio.exists"       ,  (c,cl,outer,in) ->
+        {
+          var fileIdent = new CIdent("f");
+          return CStmnt.seq(
+            CExpr.decl("FILE *", fileIdent, CExpr.call("fopen", new List<>(A0.castTo("char *"),CExpr.string("r")))),
```
Comment (fridis) [inline src/dev/flang/be/c/Intrinsics.java:150]: you `fopen` a file without ever closing it. That is not so nice, you are leaking file handles and forbid opening the file for writing.

Comment (fridis) [reply src/dev/flang/be/c/Intrinsics.java:150]: @WaelYoussfi, I do not see where this is resolved. You need to close the files.

File: src/dev/flang/be/c/Intrinsics.java [hunk 2]
```diff
@@ -143,7 +143,21 @@ public class Intrinsics extends ANY
...
+            // Testing if fopen was successful hence file/dir exists
+            CExpr.iff(CExpr.notEq(fileIdent ,new CIdent("NULL")), 
+            CExpr.iff(CExpr.eq(CExpr.call("fclose", new List<>(fileIdent)), CExpr.int8const(0)), CExpr.int8const(1).ret())),
```
Comment (WaelYoussfi) [inline src/dev/flang/be/c/Intrinsics.java:153]: @fridis I close the stream when the fopen was successful otherwise there is nothing to close.

Comment (fridis) [reply src/dev/flang/be/c/Intrinsics.java:153]: sorry, did not see this.

**Pull request merged**